src/embedded-checkout/resizable-iframe-creator.ts

```diff
@@ -0,0 +1,68 @@
+import { iframeResizer, IFrameComponent } from 'iframe-resizer';
+
+import { EmbeddedCheckoutEventType } from './embedded-checkout-events';
+import { NotEmbeddableError } from './errors';
+import parseOrigin from './parse-origin';
+
+export default class ResizableIframeCreator {
+    constructor(
+        private _options?: { timeout: number }
+    ) {}
+
+    createFrame(src: string, containerId: string): Promise<IFrameComponent> {
+        const container = document.getElementById(containerId);
+        const { timeout = 5000 } = this._options || {};
+
+        if (!container) {
+            throw new NotEmbeddableError('Unable to embed the iframe because the container element could not be found.');
+        }
+
+        const iframe = document.createElement('iframe');
+
+        iframe.src = src;
+        iframe.style.border = 'none';
+        iframe.style.display = 'none';
+        iframe.style.width = '100%';
+
+        container.appendChild(iframe);
+
+        return this._toResizableFrame(iframe, timeout)
+            .catch(error => {
+                container.removeChild(iframe);
```

Comment:
I can see that on error we remove the iframe, so probably `EmbeddedCheckout._isAttached` should be set on success?

Also, is there any reason why you remove it on failure, as opposed to only attach it on success?

Reply:
This is because I have to attach it to the DOM tree, otherwise, the browser doesn't actually load the iframe.

Reply:
yeah that was my guess ^